src/secret.ts

```diff
@@ -0,0 +1,221 @@
+        // preWarm: (context) => context.getOrInit(PrewarmClients.SECRETS_MANAGER),
+        call: async ([input], context) => {
+          const client = context.getOrInit<AWS.SecretsManager>(
+            PrewarmClients.SECRETS_MANAGER
+          );
+          const request: Partial<AWS.SecretsManager.PutSecretValueRequest> = {
+            SecretString: input.SecretString,
+            SecretBinary: input.SecretBinary,
+          };
+          if ("SecretValue" in input) {
+            if (storeAsJson) {
+              request.SecretString = JSON.stringify(input.SecretValue);
+            } else if (typeof input.SecretValue === "string") {
+              request.SecretString = input.SecretValue;
+            } else if (Buffer.isBuffer(input.SecretValue)) {
+              request.SecretBinary = input.SecretValue;
+            }
+          }
+
+          const response = await client
+            .getSecretValue({
+              ...input,
+              SecretId: secretArn,
+            })
+            .promise();
+
+          return {
+            ...response,
+            SecretValue: response.SecretString
+              ? JSON.parse(response.SecretString)
+              : undefined,
+          };
+        },
+      },
+    });
+
+    this.getJsonValue = makeIntegration<
+      "AWS.SecretsManager.GetSecretValue",
+      (
+        input?: Omit<AWS.SecretsManager.GetSecretValueRequest, "SecretId">
+      ) => Promise<
+        AWS.SecretsManager.GetSecretValueResponse & {
+          SecretValue?: SecretData;
+        }
+      >
+    >({
+      kind: "AWS.SecretsManager.GetSecretValue",
+      native: {
+        bind: (context) => {
+          this.resource.grantRead(context.resource);
+        },
+        preWarm: initClient,
+        call: async (args, context) => {
+          const response = await getSecretValue(args, context);
+          return {
+            ...response,
+            SecretValue: response.SecretString
+              ? JSON.parse(response.SecretString)
+              : undefined,
+          };
+        },
+      },
+    });
+
+    this.getValue = makeIntegration<
+      "AWS.SecretsManager.GetSecretValue",
+      (
+        input?: Omit<AWS.SecretsManager.GetSecretValueRequest, "SecretId">
+      ) => Promise<AWS.SecretsManager.GetSecretValueResponse>
+    >({
+      kind: "AWS.SecretsManager.GetSecretValue",
+      native: {
+        bind: (context) => {
+          this.resource.grantRead(context.resource);
+        },
+        call: getSecretValue,
+      },
+    });
+
+    function initClient(context: NativePreWarmContext) {
+      return context.getOrInit(PrewarmClients.SECRETS_MANAGER);
+    }
+
+    async function getSecretValue(
+      [input]: [
+        input?: Omit<AWS.SecretsManager.GetSecretValueRequest, "SecretId">
+      ],
+      context: NativePreWarmContext
+    ) {
+      const client = context.getOrInit<AWS.SecretsManager>(
+        PrewarmClients.SECRETS_MANAGER
+      );
+      console.log(secretArn);
```

Comment:
nit: remove

Reply:
I'm debugging why local stack can't find the secret

Reply:
ohh, nice